src/app/services/views/ServiceTrayView.vue

```diff
@@ -0,0 +1,67 @@
+<template>
+  <TrayView @close="emit('tray-close')">
```

Comment:
I wasn't sure whether to comment here or not, but since https://github.com/kumahq/kuma-gui/pull/1613#discussion_r1361896688 I think you want me to comment here?

All route components should be wrapped with RouteView and AppView components.

Reply:
> All route components should be wrapped with RouteView and AppView components.

Why? It needs none of the things provided by those components plus this component isn’t meant as an independent route view. It cannot be used without the corresponding list view for which we have RouteView and AppView.

Reply:
I’m not sure about this. I think we might already potentially have a bit too much nested titles going on. Personally, I’d prefer titles to at most have two levels of nesting.

On the service list view, we currently have "Services | default | Kuma". I don’t think keeping "| Kuma" makes sense (it does on the home page). Adding the selected service via a ServiceTrayView title would result in "service-1 | Services | default | Kuma".

So maybe doing titling in the list view makes more sense to we would go to "service-1 | default | Kuma" for the tray view and "Services | default | Kuma" for the plain list view?

Reply:
The biggest thing is that we should be using RouteView/RouteTitle/AppView in the route component, not only for titling purposes. I don't know why we wouldn't want to use these components, which is half the reason I tried it out in case I was missing something (see comment below with an additional commit)

Is there a reason I'm not aware of why we aren't using these components in the route component?

Reply:
The more important questions is: why do we need them here?

If there is a reason to use them, I’m all for it.

Reply:
Sorry missed this bit:

> So that’s a standalone suggestion then? Doesn’t really seem like it has anything to do with RouteView/AppView at the moment as neither component uses either stack or KCard.

Its all about consistency and coherence.

Reply:
Did you notice that we use AppView (and also RouteView) for "wizards" (e.g. zone creation)

We use AppView there for the header/title, we should also use AppView here for the header/title.

Reply:
> We use AppView there for the header/title, we should also use AppView here for the header/title.

That example doesn’t relate to this change set because neither header nor title are used here. Not a bit of functionality from RouteView or AppView is currently needed in ServiceTrayView.

Reply:
I think the issue is that AppView isn't in the changeset, we do use a header title here, but you aren't using AppView to do that (like everywhere else in the application)

Here's some grabs of our headers/titles:

![Screenshot 2023-10-20 at 09 49 48](https://github.com/kumahq/kuma-gui/assets/554604/cacf6dbb-d9d6-47b0-bb3f-efe823c290c7)


![Screenshot 2023-10-20 at 09 50 00](https://github.com/kumahq/kuma-gui/assets/554604/430bd0f2-9677-447e-852d-1d94ab7be912)

They all should use AppView to do that.

---

Did we get a decision on whether we are changing the unique HTML title when we open the tray?

Reply:
It would’ve been really helpful to know that you’re talking about the on-page title that we’re currently passing to KSlideout and not the document title.

As we’re using KSlideout for implementing trays, we should use its mechanisms for titling purposes. What you’re proposing in our alternative implementation branch is to avoid using KSlideout’s title entirely.

That difference makes me think more strongly that it would be wrong to use AppView/RouteView here. Using KSlideout’s title mechanisms (hopefully with the availability of a `title` slot in Kongponents v9) will ensure our trays look the same across different uses of KSlideout.

@adamdehaven, do you perhaps know if there are plans for other changes to KSlideout that you already know will happen as part of the development of Kongponents v9?

Reply:
> Did we get a decision on whether we are changing the unique HTML title when we open the tray?

I’ll bring that up once we start actually reviewing the concrete service trays. For now, I’m focusing on getting the underlying framework out so following implementations aren’t blocked by it.

Reply:
Oh well I'm glad those screengrabs helped then! I mentioned this in the original list of things we use AppView for (AppView does headers, not HTML titles)

> The sub route/view uses all the same patterns we already have in the application for consistency and reuse:
> RouteView/RouteTitle: route utilities and titling
> **AppView: headers/titling**
> stack and KCard: layout and spacing

---

> That difference makes me think more strongly that it would be wrong to use AppView/RouteView here. Using KSlideout’s title mechanisms (hopefully with the availability of a title slot in Kongponents v9) will ensure our trays look the same across different uses of KSlideout.

You've already pointed out the issues you were having with using the title in KSlideout, and previous to that you didn't even want to use KSlideout to do this because of the issue you had found you built your own component instead, until I suggested that we should be using KSlideout - https://github.com/kumahq/kuma-gui/pull/1613#discussion_r1360472663

I did mention I while back that we should just put our `h*` headers inside the KSlideout content instead of the title attribute, until such time that the work in KSlideout has settled down.

> [...] I do think we are ok not using this title attribute and just using <h*>s in the default slot for our titles/headers. We can also control any icon from there aswell I think.

We've already documented the issues we have in using the KSlideout title attribute, and you've mentioned quite a few times that you've had to hack around KSlideout in order to implement what we need. 

I would much rather just not use the title attribute in KSlideout rather than hacking around it. Hacking around it means it's much more likely to break as work progresses on KSlideout, and potentially means that the people working on that need to be mindful of any hacks we are using.

> For now, I’m focusing on getting the underlying framework out so following implementations aren’t blocked by it.

As I'm mentioned offline, the implementation in this PR is super important as we want to roll this out everywhere. If we need to set the title in the application when somebody opens a tray (which I would guess we do seeing as we do that when you click on a tab in other areas of the UI), then we need to support that in the implementation here in this PR, and that means using RouteView/RouteTitle.

I'm finding it hard to understand what the big problem is in following the patterns we have in the application. Even if if were the case that offered no additional functionality to what we need to do, I would still argue that for the sake of consistency/coherence they are worth using. By doing this you can make several assumptions about the application as a whole (instead of "oh! great! <insert-solution> will work because we use RouteView in all our routes...oh wait... we didn't use it in the tray 🤦 "). In many case you just automatically do things without having to think knowing that it will be ok as the application all uses the same approach. It also makes it easier to make decisions on how to build things, we just do it the same as the rest of the application.

Reply:
I have no problem at all "following the patterns we have in the application". I do so every day. I do so in this PR, too.

This stops when it’s problematic to do so. It’s not striving for consistency to avoid using KSlideout’s title mechanisms but against it.

What you’re suggesting is to only use KSlideout’s default prop so we’re able to leverage AppView and RouteView inside it instead of KSlideout’s built-in title layout (whose problematic aspects I’m in the process of addressing so we can ignore those here).

If we can’t achieve consistent titling with our current patterns while also using KSlideout’s provided title mechanisms and layout, then our current patterns for that purpose aren’t flexible enough to satisfy our use cases.

It’s more important that various uses of KSlideout look the same and we’re unlikely to reliably achieve that by not using its title mechanisms.

Reply:
> This stops when it’s problematic to do so.

I don't understand why its problematic to use RouteView/RouteTitle and AppView? You argue that there is no need to, but leaving that aside why is it problematic? Does that commit (https://github.com/kumahq/kuma-gui/commit/dfdd7a8bc8a897a52b47d479a6d60fcc2b829b5f) I added on the end here show its not problematic? 

Where is the problem there?

Reply:
I did previously explain this [here](https://github.com/kumahq/kuma-gui/pull/1611#discussion_r1366698912):

> As we’re using KSlideout for implementing trays, we should use its mechanisms for titling purposes. What you’re proposing in our alternative implementation branch is to avoid using KSlideout’s title entirely.

and [here](https://github.com/kumahq/kuma-gui/pull/1611#discussion_r1366771022) (the message you’re replying to):

>What you’re suggesting is to only use KSlideout’s default prop so we’re able to leverage AppView and RouteView inside it instead of KSlideout’s built-in title layout (whose problematic aspects I’m in the process of addressing so we can ignore those here).

> If we can’t achieve consistent titling with our current patterns while also using KSlideout’s provided title mechanisms and layout, then our current patterns for that purpose aren’t flexible enough to satisfy our use cases.

The technical difficulty stems from the fact that both AppView and KSlideout demand being provided a title for their respective mechanisms. We can’t do both because we’d end up with two titles. We shouldn’t use AppView’s mechanism because that would mean not using KSlideout’s which I consider to be higher priority for the sake of consistent across uses of KSlideout.

So, the problematic aspect here is UI-visible titling using AppView not so much RouteView and RouteTitle.

Reply:
I don't think I'm explaining myself properly. I'm asking what technical problem is caused adding a `AppView` component (whether we use the header/title slot from it or not)?

---

Separately:

> We can’t do both because we’d end up with two titles.

Currently if we use the `title` slot from TrayView and the `:title` attribute KSlideout we would end up with two titles, what you've done here is just not use the `:title` attribute, as its an optional slot thats fine to do that. If we were to use the title slot from AppView, we would just not use the optional `:title` attribute from KSlideout.

Reply:
Worth noting there, that in using the `before-title` slot we don't use the styles specified inside KSlideout title.

So any discussion around not using AppView#title for design consistency reasons is not really a valid reason seeing as we aren't using the styles specified in KSlideout for the title in this PR.

Also worth nothing that we avoid using the title slot in KCard everywhere in our application for the very same reasons and thats never been an issue before.